contracts/PartnerProxy/Renewer/PartnerRenewerProxy.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.17;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import "../../Registrar/IBaseRegistrar.sol";
+import "@rsksmart/erc677/contracts/IERC677.sol";
+import "../../BytesUtils.sol";
+import "hardhat/console.sol";
+import "./IPartnerRenewerProxy.sol";
+import "../../Renewer/IBaseRenewer.sol";
+
+contract PartnerRenewerProxy is IPartnerRenewerProxy, Ownable {
+    IBaseRegistrar private _partnerRegistrar;
+    IBaseRenewer private _partnerRenewer;
+    IERC677 private _rif;
+    // sha3('renew(string,uint)')
+    bytes4 private constant _RENEW_SIGNATURE = 0x14b1a4fc;
+
+    using BytesUtils for bytes;
+
+    modifier onlyOnce() {
+        require(owner() == address(0), "Init: Cannot be reinitialized");
+        _;
+    }
+
+    function init(
+        address _partner,
+        IBaseRegistrar partnerRegistrar,
+        IBaseRenewer partnerRenewer,
+        IERC677 rif
+    ) external onlyOnce {
+        _transferOwnership(_partner);
+        _partnerRegistrar = partnerRegistrar;
+        _partnerRenewer = partnerRenewer;
+        _rif = rif;
+    }
+
+    function renew(string calldata name, uint256 duration) external {
+        _partnerRenewer.renew(name, duration);
+    }
+
+    // - Via ERC-677
+    /* Encoding:
+        | signature  |  4 bytes      - offset  0
+        | owner      | 20 bytes      - offset  4
+        | secret     | 32 bytes      - offest 24
+        | duration   | 32 bytes      - offset 56
+        | name       | variable size - offset 88
+    */
+
+    /// @notice ERC-677 token fallback function.
+    /// @dev Follow 'Register encoding' to execute a one-transaction regitration.
+    /// @param from token sender.
+    /// @param value amount of tokens sent.
+    /// @param data data associated with transaction.
+    /// @return true if successfull.
+    function tokenFallback(
+        address from,
+        uint256 value,
+        bytes calldata data
+    ) external returns (bool) {
+        require(msg.sender == address(_rif), "Only RIF token");
+        require(data.length > 36, "Invalid data");
```

Comment:
These hardcoded integers, if they are constants can we maybe move them to constant variables, that way what they represent can be a lot more descriptive

Reply:
Another PR maybe